core/dbt/parser/macros.py

```diff
@@ -94,11 +96,21 @@ def parse_unparsed_macros(self, base_node: UnparsedMacro) -> Iterable[Macro]:
 
             name: str = macro.name.replace(MACRO_PREFIX, "")
             node = self.parse_macro(block, base_node, name)
+
+            if getattr(get_flags(), "validate_macro_args", False):
+                node.arguments = self._extract_args(macro)
+
             # get supported_languages for materialization macro
             if block.block_type_name == "materialization":
                 node.supported_languages = get_supported_languages(macro)
             yield node
 
+    def _extract_args(self, macro) -> List[MacroArgument]:
+        try:
+            return list([MacroArgument(name=arg.name) for arg in macro.args])
+        except Exception:
```

Comment:
Do we know in what cases an exception gets raised? If so, can we add a contextual comment about it?

Reply:
No. I simply did not want to add any risk at all to this code path if, for instance, the args list might ever be `None`.